test_app/CMakeLists.txt

```diff
@@ -12,6 +12,20 @@ if("${IDF_VERSION_MAJOR}.${IDF_VERSION_MINOR}" VERSION_GREATER_EQUAL "4.4")
     list(APPEND EXTRA_COMPONENT_DIRS ../pid_ctrl ../esp_encrypted_img ../pcap ../esp_jpeg)
 endif()
 
+# 2.1. Add here if the component is compatible with IDF == v4.3
+if("${IDF_VERSION_MAJOR}.${IDF_VERSION_MINOR}" VERSION_EQUAL "4.3")
+    if(NOT ((${IDF_TARGET} MATCHES "esp32c3") OR (${IDF_TARGET} MATCHES "esp32c2")))
+        list(APPEND EXTRA_COMPONENT_DIRS ../opencv)
+    endif()
+endif()
```

Comment:
In the Readme and in idf_component.yml you claim that OpenCV is compatible with IDF 4.4 only.
* What is the limitation for other versions?
* Why do you add it to v4.3 test_app here?

Reply:
Currently we can't get OpenCV to build with IDF v5.x because OpenCV source code has some issues with `uint32_t == unsigned long`.
We'll add a patch to make it work with IDF 5.x and then update this PR.